lib/cartopy/tests/test_img_transform.py

```diff
@@ -7,7 +7,13 @@
 import numpy as np
 from numpy.testing import assert_array_equal
 import pytest
-import scipy.spatial
+
+
+try:
+    import scipy.spatial
+except ImportError:
+    pytest.skip("scipy is required for image transforms", allow_module_level=True)
```

Comment:
In here, everything except `test_regridding_with_invalid_extent` will pass providing you have at least one of `scipy` or `pykdtree` installed.  Is it worth modifying this to try both imports?  Or is it too unlikely that a developer would have `pykdtree` but not `scipy`?

I am also unclear how to add appropriate skips to the parametrization of `test_regridding_with_invalid_extent` if we did that...

Reply:
Sure, it is simple enough to do the nested try/except and put the skip only at the nested version. I also added a skip to the parameterization if scipy can't be imported there.